WordPress/Classes/Login/WordPressDotComAuthenticator.swift

```diff
@@ -9,46 +9,176 @@ import Alamofire
+            user = try await withCheckedThrowingContinuation { continuation in
+                service.getAccountDetails(success: { continuation.resume(returning: $0!) }, failure: { continuation.resume(throwing: $0!) })
+            }
+        } catch {
+            throw .fetchUser(error)
+        }
+
+        // Make sure the signed-in account matches the given `accountEmail` argument.
+        if let email = context.accountEmail(in: coreDataStack.mainContext), user.email != email {
+            throw .mismatchedEmail(expectedEmail: email)
+        }
+
+        if let defaultAccountEmail = defaultAccount?.email, defaultAccountEmail != user.email {
+            throw .alreadySignedIn(signedInAccountEmail: defaultAccountEmail)
+        }
+
+        // Save the signed-in account details (and sites) into Core Data.
+        let accountID: TaggedManagedObjectID<WPAccount>
+        do {
+            let isJetpackLogin: Bool
+            if case .jetpackSite = context {
+                isJetpackLogin = true
+            } else {
+                isJetpackLogin = false
+            }
+
+            let service = WordPressComSyncService(coreDataStack: coreDataStack)
+            accountID = try await service.syncWPCom(remoteUser: user, authToken: token, isJetpackLogin: isJetpackLogin)
+        } catch {
+            DDLogError("Failed to syncing WP.com account: \(error)")
+            throw .loadingSites(error)
+        }
+
+        // Post a notification if the current signed-in account is set as the default account.
+        // This sending notification code exists because that's what the existing login system does. We can consider
+        // removing this notification once WordPressAuthenticator is removed.
+        if case .default = context {
+            let notification = Foundation.Notification.Name(rawValue: WordPressAuthenticator.WPSigninDidFinishNotification)
+            let newAccount = try? coreDataStack.mainContext.existingObject(with: accountID)
+            NotificationCenter.default.post(name: notification, object: newAccount)
+        }
+
+        return accountID
+    }
 
-            delegate.presentLoginEpilogue(
-                in: viewController,
-                for: credentials,
-                source: .custom(source: "web-login"),
-                onDismiss: { /* Do nothing */ }
-            )
+    func present(_ error: SignInError, from viewController: UIViewController) {
+        // Show an alert for non-cancellation errors.
+        let alertMessage: String
+        switch error {
+        case let .authentication(error):
+            present(error, from: viewController)
+            return
+        case .fetchUser:
+            alertMessage = NSLocalizedString("wpComLogin.error.fetchUser", value: "Failed to load user details", comment: "Error message when failing to load user details during WordPress.com login")
```

Comment:
(nit) since this chunk of code is in-between the other, more important, parts of the code, I'd suggest to either move these to `Strings` or move the error->message mapping somewhere close to the bottom of the file.

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/23749/commits/caf7f3498c287ef9ca89eef7c133c05cd57072b8